lib/src/git.rs

```diff
@@ -2251,10 +2251,11 @@ pub enum GitPushError {
     NoSuchRemote(String),
     #[error(transparent)]
     RemoteName(#[from] GitRemoteNameError),
-    #[error("Refs in unexpected location: {0:?}")]
-    RefInUnexpectedLocation(Vec<String>),
-    #[error("Remote rejected the update of some refs (do you have permission to push to {0:?}?)")]
-    RefUpdateRejected(Vec<String>),
+    #[error("Failed to push some refs")]
+    FailedRefs {
+        unexpected_locations: Vec<String>,
+        rejected_updates: Vec<String>,
+    },
```

Comment:
It's probably better to add support for partial success by not reporting failed refs as an error. (see TODO comment in `push_branches()`)

I think we can refactor the implementation as follows:

1. make `GitPushStats` public, and add `GitPushError::PartialRefUpdate(GitPushStats)` to fix the immediate problem.
2. remove deprecated git2 code path (or port git2 impl to `GitPushStats`.
3. make `git::push_branches()`/`push_updates()` return `Result<GitPushStats, _>`, update jj's view accordingly.

I think 2 and 3 can be addressed later.